src/ring_theory/dedekind_domain/ideal.lean

```diff
@@ -208,16 +208,10 @@ lemma is_dedekind_domain_inv_iff [algebra A K] [is_fraction_ring A K] :
   is_dedekind_domain_inv A ↔
     (∀ I ≠ (⊥ : fractional_ideal A⁰ K), I * I⁻¹ = 1) :=
 begin
-  set h := fraction_ring.alg_equiv A K,
-  split; rintros hi I hI,
-  { refine fractional_ideal.map_injective h.symm.to_alg_hom h.symm.injective _,
-    rw [alg_equiv.to_alg_hom_eq_coe, inv_eq, fractional_ideal.map_mul,
-        fractional_ideal.map_one_div, fractional_ideal.map_one, ← inv_eq, hi],
-    exact fractional_ideal.map_ne_zero _ hI },
-  { refine fractional_ideal.map_injective h.to_alg_hom h.injective _,
-    rw [alg_equiv.to_alg_hom_eq_coe, inv_eq, fractional_ideal.map_mul,
-        fractional_ideal.map_one_div, fractional_ideal.map_one, ← inv_eq, hi],
-    exact fractional_ideal.map_ne_zero _ hI },
+  let h := fractional_ideal.map_equiv (fraction_ring.alg_equiv A K),
+  refine h.to_equiv.forall_congr _,
+  simp only [← h.to_equiv.apply_eq_iff_eq],
+  simp [is_dedekind_domain_inv, show ⇑h.to_equiv = h, from rfl],
```

Comment:
This seems to be a bit of an API gap.  We should probably add simp lemmas for `h.to_equiv = ↑h` and `⇑↑h =⇑h `; neither of them seem to exist yet.

cc @Vierkantor @YaelDillies is this the expected simp-normal form?

Reply:
Indeed, those should exist and I know some of them are missing. Here's an example one: [`alg_hom.to_ring_hom_eq_coe` ](https://leanprover-community.github.io/mathlib_docs/algebra/algebra/basic.html#alg_hom.to_ring_hom_eq_coe)